### DISCUSSION TITLE
Conflict with playwright getByRole

### DISCUSSION THREAD

**Discussion by akoskm** [testing-library/eslint-plugin-testing-library#880] [Q&A]
I believe this is a false positive:

![image](https://github.com/testing-library/eslint-plugin-testing-library/assets/3111627/f06b3bb6-32c4-4b3c-b541-0c04575a8827)

Here's the test for completeness

```
import { test, expect } from "@playwright/test";
import { Page } from "playwright";

test("get started link", async ({ page }: { page: Page }) => {
  await page.goto("/");

  await expect(page.getByRole("link", { name: "Test" })).toHaveAttribute(
    "href",
    "example.com",
  );
});
```

One possible way to fix this would be renaming page to screen:

```
test("get started link", async ({ page: screen }: { page: Page }) => {
```

**Comment by dsumac** [comment]
Hello,

Same problem.

Is it a planed fix for this issue ?

**Comment by Belco90** [comment]
Hi @akoskm. I'm afraid this plugin is not compatible with Playwright. Please check [the shareable configs docs section](https://github.com/testing-library/eslint-plugin-testing-library?tab=readme-ov-file#shareable-configurations) to find more about compatible frameworks.

**Reply by akoskm** [reply]
Thanks @Belco90!

**Reply by razzeee** [reply]
That's fair, but you might just have a codebase with vitest/testing library and also some smoke tests via playwright. In that case, you have both in a codebase.

**Reply by Belco90** [reply]
@razzeee Yes, so you set up ESLint to only use the plugin in your vitest tests, or exclude the playwriting ones.

**Reply by razzeee** [reply]
Went with an inline ignore in the playwright file, but not sure that's ideal. Would be nice, if it would work automatically.

**Reply by Belco90** [reply]
This plugin can't detect which testing framework is supposed to run each testing file. You know the config you use in your project for each test type, so you can exclude playwright tests with a regexp in ESLint config file.
